### PR TITLE
Rebuild windows cuda 0 1 h40b5d7

### PR DESCRIPTION
**Pull request by ericpre**
Checklist
* [x] Used a [personal fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [x] Bumped the build number (if the version is unchanged)
* [x] Reset the build number to `0` (if the version changed)
* [x] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [x] Ensured the license file is being packaged.

Other relevant information:
- Remove boost migration: not necessary here, since `boost-cpp` is added to `ignore_run_export` in order to save on number of builds!
- Patches `cdt_name` for cuda 11.0 build (thanks @jaimergp!)
- Patches gcc version to match latest versions supported by CUDA - see https://github.com/conda-forge/nvcc-feedstock/issues/51

### CODE REVIEW

File: recipe/build.bat [hunk 1]
```diff
@@ -4,7 +4,16 @@ if "%1"=="cli" (set cli=1 && set build=build_cli) else (set cli=0)
 
 if "%1"=="gui" (set gui=1 && set build=build_gui) else (set gui=0)
 
-echo cli is %cli% and gui is %gui%
+if "%cuda_compiler_version%" == "None" (set gpu=0) else (set gpu=1)
+
+echo cli is %cli%, gui is %gui% and gpu is %gpu%
+
+echo CUDA_HOME is %CUDA_HOME%
```
Comment (ericpre) [inline recipe/build.bat]: It seems that `CUDA_HOME` is not set correctly even if it should be set by nvcc, if I understand correctly?! @jaimergp would you have any idea about this?

Comment (jaimergp) [reply recipe/build.bat]: Try CUDA_PATH in Windows (although we should be setting that too).

**Comment by jaimergp** [comment]
Hm, it looks that

1. CUDA installation is finishing successfully ([link](https://dev.azure.com/conda-forge/feedstock-builds/_build/results?buildId=245352&view=logs&j=171a126d-c574-5c8c-1269-ff3b989e923d&t=11e12034-2a2b-52f8-fa15-1d2b32e6923f&l=113))
2. But then nvcc cannot be found in PATH ([link](https://dev.azure.com/conda-forge/feedstock-builds/_build/results?buildId=245352&view=logs&j=171a126d-c574-5c8c-1269-ff3b989e923d&t=1183ba29-a0b5-5324-8463-2a49ace9e213&l=294))

**Comment by ericpre** [comment]
Indeed, the log says:
```
(base) D:\bld\prismatic_split_1606423578869\work>if not defined CUDA_PATH (
for /F "usebackq tokens=*" %a in (`where nvcc.exe`) do set "CUDA_NVCC_EXECUTABLE=%a"   || goto :error  
 if "" == "" (
echo "Cannot determine CUDA_PATH: nvcc.exe not in PATH"  
 exit /b 1 
)  else (for /F "usebackq tokens=*" %a in (`python -c "from pathlib import Path; print(Path('').parents[1])"`) do set "CUDA_PATH=%a"   || goto :error ) 
) 
INFO: Could not find files for the given pattern(s).
"Cannot determine CUDA_PATH: nvcc.exe not in PATH"

```

**Comment by jaimergp** [comment]
Ah, I found the bug! It's a problem in our `nvcc` wrapper activation. I'll submit a PR so you will have a fix soon.

In the meantime, you can set the CUDA root dir to `C:\Program Files\NVIDIA GPU Computing Toolkit\CUDA\v%cuda_compiler_version%` or something similar.

**Comment by jaimergp** [comment]
Oh, yeah, [this error](https://dev.azure.com/conda-forge/feedstock-builds/_build/results?buildId=245371&view=logs&j=e5cdccbf-4751-5a24-7406-185c9d30d021&t=66e4a587-a723-53e9-873c-6c9a2c5971ea&l=421). CMake expect forward slashes only and you need to quote the full thing. Use this var expansion:

```
set "CUDA_HOME=C:\Program Files\NVIDIA GPU Computing Toolkit\CUDA\v%cuda_compiler_version%"
set "CUDA_HOME=%CUDA_HOME:\=/%"
```

**Comment by jaimergp** [comment]
I also realized you might need this! Sorry I didn't notice earlier:

https://docs.conda.io/projects/conda-build/en/latest/user-guide/environment-variables.html#inherited-environment-variables

Edit: I am not using that on `openmm-feedstock` and works fine so no clue :/

**Comment by ericpre** [comment]
OK, I will have a look at it tomorrow. Thanks!

**Comment by jaimergp** [comment]
We are going to try a different thing now...

1. Get `run_conda_forge_build_setup_win.bat` and `install_cuda.bat` from [this fork](https://github.com/jaimergp/conda-forge-ci-setup-feedstock/tree/windows-cuda-driver/recipe) (part of [this PR](https://github.com/conda-forge/conda-forge-ci-setup-feedstock/pull/131))
2. Copy them to this PR, under `recipe`
3. Re-render.
4. 🤞 to see if it works!

What we are going to do is to temporarily replace the `conda-forge-ci-setup` parts for Windows and see if that fixes this behavior. This is working on OpenMM, so it *should* work here too!

Thanks for your patience!

**Comment by ericpre** [comment]
OK, let see if this works!

**Comment by jaimergp** [comment]
I am honestly puzzled this does not work. All I can see is that upstream uses a [deprecated CMake module to find CUDA](https://cmake.org/cmake/help/latest/module/FindCUDA.html). OpenMM does use the recommended replacement, so maybe that's something to deal with upstream? Sorry I can't help more...

**Comment by jaimergp** [comment]
Hm, last build for OpenMM was using CMake 3.18, but this is using 3.19. Might be worth trying the change?

**Comment by ericpre** [comment]
The `CUDA_HOME` is now defined when calling the build script, so your last fixes are working:
```
cli is 1 , gui is 0 and gpu is 1
CUDA_HOME is C:/Program Files/NVIDIA GPU Computing Toolkit/CUDA/v10.2
CMake prefix path: D:\bld\prismatic_split_1606471570432\_h_env\Library
```

> I am honestly puzzled this does not work. All I can see is that upstream uses a [deprecated CMake module to find CUDA](https://cmake.org/cmake/help/latest/module/FindCUDA.html). OpenMM does use the recommended replacement, so maybe that's something to deal with upstream? Sorry I can't help more...

I agree that this may be the issue here and I will patch the cmake config file to update finding CUDA. I had a version of this recipe which was working to build locally on windows, where I set a few couple of environment variables and I should be able to work out something!

**Comment by ericpre** [comment]
Actually, I was wrong this morning: `CUDA_HOME` and `CUDA_PATH` are not set correctly in https://github.com/conda-forge/prismatic_split-feedstock/pull/17/commits/9d66dc0751e90c911e41d88247389190ab9879cd after using the local script.
Setting `CUDA_PATH` manually works just fine.

Once the current build is finished, I will remove the local scripts to check if it makes any difference.

**Comment by ericpre** [comment]
To summarise and clarify the outcome of this PR: 
- building with cuda on windows works well :tada: 🍾 :confetti_ball:
- `CUDA_HOME` and `CUDA_PATH` are not defined.

We tried to use local script aiming to fix `CUDA_HOME` and `CUDA_PATH` which are supposedly set by `nvcc` package (in the activation script). The log of https://github.com/conda-forge/prismatic_split-feedstock/pull/17/commits/446325fb6f46d700b900f80b14bbcdfc9da7e6fc shows that this isn't working as expected in this PR, the following shows some output of the [build script](https://github.com/conda-forge/prismatic_split-feedstock/blob/696d41fd8ff448624744856b332f4370fd4275da/recipe/build.bat#L9-L16):
```
cli is 1 , gui is 0 and gpu is 1
CUDA_HOME: 
CUDA_PATH: 
CUDA_PATH (after): C:\Program Files\NVIDIA GPU Computing Toolkit\CUDA\v10.2
CMake prefix path: D:\bld\prismatic_split_1606515158327\_h_env\Library
```

To confirm, the log of https://github.com/conda-forge/prismatic_split-feedstock/pull/17/commits/3c96112cd8e8b14be250b9d4eddff98dd50fc07e where the local script have been removed shows the same:
```
cli is 1 , gui is 0 and gpu is 1
CUDA_HOME: 
CUDA_PATH: 
CUDA_PATH (after): C:\Program Files\NVIDIA GPU Computing Toolkit\CUDA\v10.2
CMake prefix path: D:\bld\prismatic_split_1606515158327\_h_env\Library
```

In both case, the `CUDA_PATH` have been set manually in the build script so that cmake can find CUDA and the build can work. In this case, there is no need to define `CUDA_TOOLKIT_ROOT_DIR` - this is still using the deprecated `FindCUDA` cmake module.

Thanks @jaimergp for getting building cuda on windows to work on conda-forge!

**Comment by jaimergp** [comment]
Thanks for the summary and glad you got it working!

I'll investigate why the env vars do not reach the build script in this feedstock. Might have to do with multi-outputs.

**Pull request merged**